src/DominantJuice.sol

```diff
@@ -0,0 +1,399 @@
+     * to receive a `didRedeem()` call. `JBPayoutRedemptionPaymentTerminal3_1_1.redeemTokensOf()` calls
+     * `JBSingleTokenPaymentTerminalStore3_1_1.recordRedemptionFor()`, which calls this function.
+     * @param _data JB project redemption data. See https://docs.juicebox.money/dev/api/data-structures/jbredeemparamsdata/.
+     * @return reclaimAmount Amount to be reclaimed from the treasury. This is useful for optionally customizing how much
+     * funds from the treasury are disbursed per redemption.
+     * @return memo A memo to be forwarded to the event. Useful for describing any new actions are being taken.
+     * @return delegateAllocations Amount to be sent to delegates instead of being added to the beneficiary. Useful for
+     * auto-routing funds from a treasury as redemptions are sought.
+     */
+    function redeemParams(JBRedeemParamsData calldata _data)
+        external
+        view
+        virtual
+        override
+        redeemCycleCheck
+        returns (
+            uint256 reclaimAmount,
+            string memory memo,
+            JBRedemptionDelegateAllocation3_1_1[] memory delegateAllocations
+        )
+    {
+        address payable redeemer = payable(_data.holder);
+
+        // The if statements are the only changes from the redeemParams() template.
+        if (unwrap(pledgers.pledgerWeight[redeemer]) == 0) revert MustBePledger();
+        if (pledgers.hasBeenRefunded[redeemer]) revert AlreadyWithdrawnRefund();
+
+        // Forward the default reclaimAmount received from the protocol.
+        reclaimAmount = _data.reclaimAmount.value;
+        // Forward the default memo received from the redeemer.
+        memo = _data.memo;
+        // Add `this` contract as a Redeem Delegate so that it receives a `didRedeem` call. Don't send any extra funds to the delegate.
+        delegateAllocations = new JBRedemptionDelegateAllocation3_1_1[](1);
+        delegateAllocations[0] = JBRedemptionDelegateAllocation3_1_1(this, 0, "");
+    }
+
+    /**
+     * @notice Received hook from the payment terminal after a redemption. If cycle meets project target,
+     * then this function cannot be called, since pauseRedemptions will be true. If the cycle fails to meet
+     * its target, then this function will be callable for pledgers after the `cycleExpiryDate`.
+     * @dev Reverts if the calling contract is not one of the project's terminals. This is the final call in the
+     * call chain described in `redeemParams()`.
+     * @param _data Standard Juicebox project redemption data.
+     * See https://docs.juicebox.money/dev/api/data-structures/jbdidredeemdata3_1_1/.
+     */
+    function didRedeem(JBDidRedeemData3_1_1 calldata _data) external payable terminalCheck redeemCycleCheck {
+        if (_data.projectId != campaign.projectId) revert IncorrectProjectID();
+
+        address payable redeemer = payable(_data.holder);
```

Comment:
would rename to `pledger` or `pledgerAddress` - cognitive load to map `redeemer` to `pledger` in your mind

Reply:
Since you’re my gracious sensei, I’ve changed all redeemer instances to pledger. (For what it’s worth, in my brain, `pledger.pledgerX[pledgerY].pledger` makes me feel like I’m chest deep in the waters of sameness and puts me to sleep, haha. I appreciate cognitive load because it keeps me on my toes, and the mental work helps with information retention.) But again, you are the boss here!

Reply:
haha, I totally get that! And you could argue that having it be redeemer is more correct since it may or may not be a pledger....just someone who's trying to redeem.